src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php

```diff
@@ -130,14 +130,14 @@ public function reverseTransform($value)
         try {
             if ($dateOnly) {
                 // we only care about year-month-date, which has been delivered as a timestamp pointing to UTC midnight
-                return new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->inputTimezone));
+                $dateTime = new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->outputTimezone));
+            } else {
+                // read timestamp into DateTime object - the formatter delivers a timestamp
+                $dateTime = new \DateTime(sprintf('@%s', $timestamp));
+                // set timezone separately, as it would be ignored if set via the constructor,
+                // see http://php.net/manual/en/datetime.construct.php
+                $dateTime->setTimezone(new \DateTimeZone($this->outputTimezone));
```

Comment:
passing a TZ as constructor does *not* "set" it.
To set it, you need to call "setTimezone".
passing a TZ as constructor does *only* tell the constructor that *if* the first arg contains no TZ info, then this should be the one to use while parsing.
That's why